conda_libmamba_solver/solver.py

```diff
@@ -1004,8 +1004,15 @@ def _notify_conda_outdated(
             # This check can be silenced with a specific option in the context or in quiet mode
             return
 
-        current_conda_prefix_rec = PrefixData(context.conda_prefix).get("conda", None)
-        if not current_conda_prefix_rec:
+        # manually check base prefix since `PrefixData(...).get("conda", None) is expensive
+        # once prefix data is lazy this might be a different situation
+        current_conda_prefix_rec = None
+        for filename in os.scandir(os.path.join(context.conda_prefix, "conda-meta")):
+            if filename.name.endswith(".json") and filename.name.rsplit("-", 2)[0] == "conda":
```

Comment:
I think `pathlib.Path.glob()` uses scandir under the hood, so maybe we can get away with the more readable:


```suggestion
        for filename in Path(context.conda_prefix, "conda-meta").glob("conda-*.json"):
            if filename.name.rsplit("-", 2)[0] == "conda":
```

Reply:
Ah, only on recent Python versions. Ugh!